src/main/java/com/teragrep/lsh_01/MessageProcessor.java

```diff
@@ -87,19 +87,39 @@ public void run() {
                 LOGGER.debug("Healthcheck endpoint called");
                 response = generateResponse(messageHandler.responseHeaders());
             }
-            else if (isTokenOk()) {
-                LOGGER.debug("Processing message");
-                response = processMessage();
-            }
-            else if (!req.headers().contains(HttpHeaderNames.AUTHORIZATION)) {
-                LOGGER.debug("Required authorization not provided; requesting authentication.");
-                response = generateAuthenticationRequestResponse();
-            }
             else {
-                LOGGER.debug("Invalid authorization; rejecting request.");
-                response = generateFailedResponse(HttpResponseStatus.UNAUTHORIZED);
+                if (messageHandler.requiresToken()) {
+                    if (req.headers().contains(HttpHeaderNames.AUTHORIZATION)) {
+                        HttpResponse response1;
+                        try {
+                            Subject subject = messageHandler
+                                    .asSubject(req.headers().get(HttpHeaderNames.AUTHORIZATION));
+                            req.headers().remove(HttpHeaderNames.AUTHORIZATION);
```

Comment:
add comment here that it's removed to prevent leak

Reply:
Fixed